### PR TITLE
Updates to curation panel

### PR DESCRIPTION
**Pull request by calvinlu3**
This fixes https://github.com/oncokb/oncokb/issues/3136

### CODE REVIEW

File: src/main/webapp/app/shared/button/CurationModeButton.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+interface CurationModeButtonProps {
+  curateEntityId: string | number;
+}
+
+const CurationModeButton: React.FunctionComponent<
+  CurationModeButtonProps & StoreProps & ButtonProps & React.HTMLAttributes<HTMLButtonElement>
+> = props => {
+  const { toggleCurationPanel, curateEntityId, ...buttonProps } = props;
+  return (
+    <Button tag={Link} to={`/fda-submission/${curateEntityId}/curate`} color="secondary" type="button" {...buttonProps}>
```
Comment (zhx828) [inline src/main/webapp/app/shared/button/CurationModeButton.tsx:17]: `/fda-submission/` should be passed through property. We could create entity enum and create route utils to return proper URL based on entity and action. Action should be enum as well.

**Review by zhx828 — CHANGES_REQUESTED** [review]
- [ ] No need to uppercase `CURATION PANEL`
- [ ] Please check all places using SaveButton. Some of the properties are lost, especially the `disabled`

**Comment by calvinlu3** [comment]
Ended up removing the `CurateButton` and merged the logic into `EntityActionButton` (which handles view, edit, curate, delete, and create).

**Review by zhx828 — APPROVED** [review]
Nicely done 💯

**Pull request merged**